File: HyLang/Parser/AST/BinaryNode.cs
namespace HyLang.Parser.AST;

public class BinaryNode:Node
{
    public NumberNode left;
    public NumberNode right;
    public string operation;

    public BinaryNode(NumberNode left, string operation, NumberNode right)
    {
        this.left = left;
        this.operation = operation;
        this.right = right;
    }
}

File: src/parser/expression/binary_node.rs
use crate::{lexer::token::Token, parser::node::{DNode, Node}};

#[derive(Debug)]
pub struct BinaryNode {
    token: Token,
    left: DNode,
    right: DNode,
    op: BinaryOp,
}

impl BinaryNode {
    pub fn new(token: Token, left: DNode, right: DNode, op: BinaryOp) -> Box<Self> {
        Box::new(Self { token, left, right, op })
    }
}

impl Node for BinaryNode {
    fn visit(&self) {
        todo!()
    }
}

#[derive(Debug)]
pub enum BinaryOp {
    Add,
    Sub,
    Mult,
    Dev,
}
